diff -urN chicken-5.2.0.orig/chicken.h chicken-5.2.0/chicken.h
--- chicken-5.2.0.orig/chicken.h	2020-02-29 14:41:06.000000000 +0100
+++ chicken-5.2.0/chicken.h	2020-09-27 14:33:57.683954279 +0200
@@ -1483,17 +1483,17 @@
 #ifdef C_BIG_ENDIAN
 # ifdef C_SIXTY_FOUR
 #  define C_lihdr(x, y, z)              ((C_LAMBDA_INFO_TYPE >> 56) & 0xff), \
-                                        0, 0, 0, 0, (x), (y), (z)
+                                        0, 0, 0, 0, (x), (y), ((C_char)(z))
 # else
 #  define C_lihdr(x, y, z)              ((C_LAMBDA_INFO_TYPE >> 24) & 0xff), \
-                                        (x), (y), (z)
+                                        (x), (y), ((C_char)(z))
 # endif
 #else
 # ifdef C_SIXTY_FOUR
-#  define C_lihdr(x, y, z)              (z), (y), (x), 0, 0, 0, 0, \
+#  define C_lihdr(x, y, z)              ((C_char)(z)), (y), (x), 0, 0, 0, 0, \
                                         ((C_LAMBDA_INFO_TYPE >> 56) & 0xff)
 # else
-#  define C_lihdr(x, y, z)              (z), (y), (x), \
+#  define C_lihdr(x, y, z)              ((C_char)(z)), (y), (x), \
                                         ((C_LAMBDA_INFO_TYPE >> 24) & 0xff)
 # endif
 #endif
